### PR TITLE
Add ability to call outside endpoints for adaptive authentication

### PR DESCRIPTION
**Pull request by madurangasiriwardena**
### Proposed changes in this pull request

- Add ability to call outside endpoints for adaptive authentication

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/LongWaitStatus.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+package org.wso2.carbon.identity.application.authentication.framework.model;
+
+/**
+ * Object holding long wait status.
+ */
+public class LongWaitStatus {
+
+    public enum Status {
+        WAITING, COMPLETED
```
Comment (pulasthi7) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/LongWaitStatus.java]: We need to handle time outs also, right?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/GraphBasedSequenceHandler.java [hunk 2]
```diff
@@ -277,20 +360,61 @@ private void handleDecisionPoint(HttpServletRequest request, HttpServletResponse
...
+     * This method allows a BiConsumer which throws exceptions to be used in places which expects a BiConsumer.
+     *
+     * @param <T>         the type of the input to the function
+     * @param <U>         the type of the input to the function
+     * @param <V>         the type of the input to the function
+     * @param <E>         the type of Exception
+     * @param triConsumer instances of the {@code TriConsumerWithExceptions} functional interface
+     * @return an instance of the {@code BiConsumer}
+     */
+    public static <T, U , V , E extends Exception> AsyncReturn rethrowTriConsumer(AsyncReturn triConsumer) {
```
Comment (pulasthi7) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/sequence/impl/GraphBasedSequenceHandler.java]: Shouldn't this be AsyncReturn<T, U , V , E extends Exception> ?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java [hunk 3]
```diff
@@ -253,7 +251,14 @@ public static ApplicationAuthenticator getAppAuthenticatorByName(String name) {
     public static AuthenticationContext getContextData(HttpServletRequest request) {
 
         AuthenticationContext context = null;
-
+        if(request.getParameter("promptResp") != null && request.getParameter("promptId") != null){
```
Comment (pulasthi7) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java]: formatting

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/LongWaitStatus.java (hunk 1 above)
Comment (madurangasiriwardena) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/LongWaitStatus.java]: Created https://github.com/wso2/product-is/issues/3188 to track the improvement.

**Pull request merged**